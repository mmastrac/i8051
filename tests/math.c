/* test8051.c â€” SDCC 8051 test program */

#include <8051.h>  /* gives you SFR definitions like P1 */


void main(void)
{
    __xdata unsigned char *scratch = (__xdata unsigned char *)0x8000;

    volatile int a = 10;
    volatile int b = a * 25;
    volatile int c = b / 10;
    volatile int d = (c + 1) % 10;

    /* Store result in external memory for visibility */
    scratch[0] = a & 0xff;
    scratch[1] = b & 0xff;
    scratch[2] = c & 0xff;
    scratch[3] = d & 0xff;

    /* Stop here (endless loop) */
    while (1) {
        /* idle */
    }
}
