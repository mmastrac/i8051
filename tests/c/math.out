CPU halted at 0x0101 after 992 instructions
  A=06  B=FC  DPTR=8003  C=false OV=false AC=false F0=false
  R0=00 R1=00 R2=00 R3=00 R4=00 R5=00 R6=00 R7=06 
MEM: 0A FA 19 06
